packages/react-reconciler/src/ReactFiberRootScheduler.js

```diff
@@ -229,13 +229,17 @@ function flushSyncWorkAcrossRoots_impl(
   isFlushingWork = false;
 }
 
-function processRootScheduleInMicrotask() {
+function processRootScheduleInImmediateTask() {
   if (enableProfilerTimer && enableComponentPerformanceTrack) {
     // Track the currently executing event if there is one so we can ignore this
     // event when logging events.
     trackSchedulerEvent();
```

Comment:
I had to restructure this a bit so that we have two separate callbacks depending on if we do the `Scheduler_scheduleCallback` pass or the microtask pass. Because we shouldn't call `trackSchedulerEvent()` if we're inside a microtask callback because that could mark a real event as if it was a scheduler event (and therefore hidden).